### PR TITLE
Codecoverage

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY

Code coverage support for unit and integration tests. 

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

**Comment by gundalow** [comment]
`ansible-test coverage` is failing with
`ERROR: 'CoverageData' object has no attribute 'read_file'`

This maybe an issue in my setup, or in ansible-test
Tracking as https://github.com/ansible/ansible/issues/65907

File: tests/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+if find tests/output/coverage/ -mindepth 1 -name '.*' -prune -o -print -quit | grep -q .; then
+    stub=""
+
+    # shellcheck disable=SC2086
+    ansible-test coverage xml -v --requirements --group-by command --group-by version ${stub:+"$stub"}
+
+    # upload coverage report to codecov.io
+    for file in tests/output/coverage/coverage=*.xml; do
```
Comment (seuf) [inline tests/coverage.sh]: can you add a comment with an example of file name parsed here ?

File: tests/coverage.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        flags="${file##*/coverage=}"
+        flags="${flags%.xml}"
+        flags="${flags//=/,}"
+        flags="${flags//[^a-zA-Z0-9_,]/_}"
+
+        bash <(curl -s https://codecov.io/bash) \
+            -f "${file}" \
+            -F "${flags}" \
+            -t 83cd8957-dc76-488c-9ada-210dcea51633 \
```
Comment (mattclay) [inline tests/coverage.sh]: Arbitrary collections shouldn't be uploading code coverage for the ansible/ansible repo.

Comment (gundalow) [reply tests/coverage.sh]: Ah, I thought I had corrected the token, thank you.

File: tests/coverage.sh (hunk 1 above)
Comment (gundalow) [reply tests/coverage.sh]: Good idea, added

**Comment by gundalow** [comment]
Ready for review 

Once merged the overview for `master` can be found at https://codecov.io/gh/ansible-collections/grafana
This will work once this PR has been merged: 

https://github.com/ansible-collections/grafana/pull/20#issuecomment-568191520

**Pull request merged**